lib/CGameInfoCallback.cpp

```diff
@@ -282,6 +292,9 @@ bool CGameInfoCallback::getHeroInfo(const CGObjectInstance * hero, InfoAboutHero
 
 	dest.initFromHero(h, accessFlag);
 
+	if (accessFlag && !gs->curB)
+		dest.details->manaLimit = -1; //we do not want to leak max mana info outside battle so set to meaningless value
```

Comment:
This is of course not how something like that have to be implemented.
Instead you need extend InfoAboutHero's "detailed" bool and probably replace it with enum like:
EInfoLevel: BASIC, DETAILED, INBATTLE

This would let us extend info level if sometimes in future we'll have more things depend on it.